source/backend/dal/Repositories/PropertyRepository.cs

```diff
@@ -406,68 +407,79 @@ public PimsProperty TransferFileProperty(PimsProperty property, bool isOwned, bo
                     p.PimsPropPropRoadTypes.Any(pl => filter.TenureRoadTypes.Contains(pl.PropertyRoadTypeCode)));
             }
 
             // Lease filters
             if (!string.IsNullOrEmpty(filter.LeaseStatus))
             {
-                query = query.Where(p =>
+                predicate.And(p =>
                     p.PimsPropertyLeases.Any(pl => pl.Lease.LeaseStatusTypeCode == filter.LeaseStatus));
             }
 
             if (filter.LeaseTypes != null && filter.LeaseTypes.Count > 0)
             {
-                query = query.Where(p =>
+                predicate.And(p =>
                     p.PimsPropertyLeases.Any(pl => filter.LeaseTypes.Contains(pl.Lease.LeaseLicenseTypeCode)));
             }
 
             if (filter.LeasePurposes != null && filter.LeasePurposes.Count > 0)
             {
-                query = query.Where(p =>
+                predicate.And(p =>
                     p.PimsPropertyLeases.Any(pl => filter.LeasePurposes.Contains(pl.Lease.LeasePurposeTypeCode)));
             }
 
             if (!string.IsNullOrEmpty(filter.LeasePayRcvblType))
             {
-                query = query.Where(p =>
-                    p.PimsPropertyLeases.Any(pl => (pl.Lease.LeasePayRvblTypeCode == filter.LeasePayRcvblType || filter.LeasePayRcvblType == "all") && (pl.Lease.OrigExpiryDate >= DateTime.Now.Date || pl.Lease.PimsLeaseTerms.Any(t => t.TermExpiryDate == null || t.TermExpiryDate >= DateTime.Now.Date))));
+                // Check the status
+                predicate.And(p => p.PimsPropertyLeases.Any(pl => pl.Lease.LeasePayRvblTypeCode == filter.LeasePayRcvblType || filter.LeasePayRcvblType == "all"));
+
+                // Check not expired
+                predicate.And(p => p.PimsPropertyLeases.Any(pl => (pl.Lease.PimsLeaseTerms.Any(t => !t.TermExpiryDate.HasValue) // any term expiry is null
+                                        || (!pl.Lease.PimsLeaseTerms.Any() && !pl.Lease.OrigExpiryDate.HasValue) // no terms and orig is null
+                                        || (!pl.Lease.PimsLeaseTerms.Any() && pl.Lease.OrigExpiryDate.HasValue && pl.Lease.OrigExpiryDate.Value.Date >= DateTime.Now.Date) // no terms and orig not expired
+                                        || (pl.Lease.OrigExpiryDate.HasValue // has terms and term expired is bigger than Lease expiry is not expired
+                                            && pl.Lease.PimsLeaseTerms.Any(lt => lt.TermExpiryDate.HasValue && lt.TermExpiryDate.Value.Date > pl.Lease.OrigExpiryDate.Value.Date && lt.TermExpiryDate.Value.Date >= DateTime.Now.Date))
+                                        || (pl.Lease.OrigExpiryDate.HasValue // has terms and term expired is lower than Lease expiry
+                                            && (!pl.Lease.PimsLeaseTerms.Any(lt => lt.TermExpiryDate.HasValue && lt.TermExpiryDate.Value.Date > pl.Lease.OrigExpiryDate.Value.Date) && pl.Lease.OrigExpiryDate.Value.Date >= DateTime.Now.Date)))));
```

Comment:
I see the old logic is using **DateTime.Now.Date** but I wonder if we should be using **DateTime.UtcNow.Date** instead given that we store date-time in UTC in the database. Otherwise this code might return different results based on server timezone (PST vs UTC)

What do you think @devinleighsmith ?